src/rpc/client.cpp

```diff
@@ -15,118 +16,245 @@
+    { "clearbanned",                 {{}, {}} },
+    // rawtransaction
+    { "getrawtransaction",           {{s}, {o, s}} },
+    { "gettxoutproof",               {{o}, {s}} },
+    { "verifytxoutproof",            {{s}, {}} },
+    { "createrawtransaction",        {{o, o}, {o, o}} },
+    { "decoderawtransaction",        {{s}, {}} },
+    { "decodescript",                {{s}, {}} },
+    { "signrawtransaction",          {{s}, {o, o, s, s}} },
+    { "sendrawtransaction",          {{s}, {o}} },
+    // rpcdisclosure
+    { "z_getpaymentdisclosure",      {{s, o, o}, {s}} },
+    { "z_validatepaymentdisclosure", {{s}, {}} },
+    // rpcdump
+    { "importprivkey",               {{s}, {s, o}} },
+    { "importaddress",               {{s}, {s, o, o}} },
+    { "importpubkey",                {{s}, {s, o}} },
+    { "z_importwallet",              {{s}, {}} },
+    { "importwallet",                {{s}, {}} },
+    { "dumpprivkey",                 {{s}, {}} },
+    { "z_exportwallet",              {{s}, {}} },
+    { "z_importkey",                 {{s}, {s, o}} },
+    { "z_importviewingkey",          {{s}, {s, o}} },
+    { "z_exportkey",                 {{s}, {}} },
+    { "z_exportviewingkey",          {{s}, {}} },
+    // rpcwallet
+    { "getnewaddress",               {{}, {s}} },
+    { "getrawchangeaddress",         {{}, {}} },
+    { "sendtoaddress",               {{s, o}, {s, s, o}} },
+    { "listaddresses",               {{}, {}} },
+    { "listaddressgroupings",        {{}, {o}} },
+    { "signmessage",                 {{s, s}, {}} },
+    { "getreceivedbyaddress",        {{s}, {o, o, o}} },
+    { "getbalance",                  {{}, {s, o, o, o, o}} },
+    { "sendmany",                    {{s, o}, {o, s, o}} },
+    { "addmultisigaddress",          {{o, o}, {s}} },
+    { "listreceivedbyaddress",       {{}, {o, o, o, s, o, o}} },
+    { "listtransactions",            {{}, {s, o, o, o, o}} },
+    { "listsinceblock",              {{}, {s, o, o, o, o, o}} },
+    { "gettransaction",              {{s}, {o, o, o}} },
+    { "backupwallet",                {{s}, {}} },
+    { "keypoolrefill",               {{}, {o}} },
+    { "walletpassphrase",            {{s, o}, {}} },
+    { "walletpassphrasechange",      {{s, s}, {}} },
+    { "walletconfirmbackup",         {{s}, {}} },
+    { "walletlock",                  {{}, {}} },
+    { "encryptwallet",               {{s}, {}} },
+    { "lockunspent",                 {{o, o}, {}} },
+    { "listlockunspent",             {{}, {}} },
+    { "settxfee",                    {{o}, {}} },
+    { "getwalletinfo",               {{}, {o}} },
+    { "resendwallettransactions",    {{}, {}} },
+    { "listunspent",                 {{}, {o, o, o, o, o, o}} },
+    { "z_listunspent",               {{}, {o, o, o, o, o}} },
+    { "fundrawtransaction",          {{s}, {o}} },
+    { "zcsamplejoinsplit",           {{}, {}} },
+    { "zcbenchmark",                 {{s, o}, {}} },
```

Comment:
There is at least one other case somewhere that can be either a string or some other type. Currently, that means for those cases we _always_ convert, and so strings need to be passed as `"\"some value\""`. I had asked somewhere whether it would make sense to have a third case where we _attempt_ to convert, but send the string on failure, which wouldn’t break uses that were already sending `"\"some value\""`, but would then also allow `"some value"` (similar to the `getblockdeltas` change in this PR).

Reply:
I notice that we were previously converting the third argument. I’m going to preserve that behavior here (so, requires `"\"some value\""` in the string case) and I opened #6543 to support the optional conversion.